### ISSUE TITLE
NxEnhanced Userscript not working via Adguard Extensions Component

### ISSUE THREAD

**Issue by jenabaivab** [AdguardTeam/CoreLibs#1360]
**Steps to reproduce**

1.  Install NxEnhanced Script (https://github.com/hjk789/NXEnhanced) from greasyfork.
2.  Check results on NextDNS interface.
3.  No new features appear.

**Expected behavior**

It should show the extra features that the script provides on the NextDNS logs page like Allow/Deny per domain etc.

**Actual behavior**

None of the extra features are visible.

I tried using this script with ViolentMonkey and it works perfectly.

**Your environment**

♦   Environment name and version: Firefox 82.0b9
♦   Operating system and version: Windows 10 build 19041.388

**Comment by hjk789** [comment]
Hey, there @jenabaivab ! I'm NX Enhanced's developer. I just now stumbled up on this page. Are you still having this problem on Firefox?

**Comment by hjk789** [comment]
@jenabaivab  The Firefox extension is now available on Firefox Add-ons. Check it out: https://addons.mozilla.org/addon/nx-enhanced/

**Comment by jenabaivab** [comment]
@hjk789 that's awesome. Unfortunately, I have stopped using nextdns since I always get connected to Singapore and Amsterdam even though there are 4 servers in India.  I have and will continue recommending NXEnhanced to anyone using NextDNS for sure,

**Comment by hjk789** [comment]
Yeah, their routing is kinda... "confused". There seems to be no clear logic for the servers it chooses. But I understand. And thanks very much for your support!

**Comment by wanesoft** [comment]
Hi, @hjk789 good to see you! So. After a small research, I only see the option to import/export settings and whitelist/blacklist sorting. Unfortunately, nothing changes in the Logs tab. Regarding script injection: no difference is observed when using the Adguard or Tampermonkey. Can you check it with me? My configurations: firefox+tampermonkey, firefox+greasemonkey, firefox+adguard, chrome+tampermonkey and chrome+adguard.

**Comment by hjk789** [comment]
@wanesoft In Firefox, the logs features don't load for two reasons: 1. There's a badly coded condition right in the beginning of the userscript that prevents the features from being added to the page (this is fixed in the extension). And 2. In Firefox there's a CSP error when injecting code in the frame used to contact the server's API, and the only way I've found to workaround this is by disabling CSP in about:config, but that's not recommended, which is why I created the Firefox extension.   
In Chrome it should work, but I just tested here and it's not working too, but when I removed the condition and replaced with the new code it loaded fine.   
I'll release an update for the userscript with some of the fixes and improvements implemented in the extension. But in the meantime, to know whether it's a problem in the script or in the injection, click on the "Export this config" button in the Settings page, if it generates a JSON file then it's the script fault, but if the spinner keeps spinning forever, then something else is happening.
